### PR TITLE
Use urllib in koji to_compose conversion instead of requests

### PR DESCRIPTION
**Pull request by velezd**
This allows us to use ftp for compose url

### CODE REVIEW

File: libpermian/plugins/koji/__init__.py [hunk 1]
```diff
@@ -88,18 +95,21 @@ def to_compose(self):
...
-                except RuntimeError: # raised by productmd when failed to load compose metadata
-                    pass
+            try:
+                with urllib.request.urlopen(entrypoint, timeout=5) as response:
+                    compose_relpaths = response.read().decode().strip()
+                    compose_relpath = compose_relpaths.split('\n')[-1]
+                    compose_path = f'{entrypoint_dir}/{compose_relpath}'
+                    try:
+                        compose_id = productmd.compose.Compose(compose_path).info.compose.id
+                        return ComposeStructure(self.settings, compose_id, location=compose_path)
```
Comment (pholica) [inline libpermian/plugins/koji/__init__.py:109]: It would be great to add LOGGER.debug printing compose path and compose id before returning it.

**Review by pholica — CHANGES_REQUESTED** [review]

File: libpermian/plugins/koji/__init__.py (hunk 1 above)
Comment (pholica) [reply libpermian/plugins/koji/__init__.py:109]: Otherwise, it looks good.

Comment (velezd) [reply libpermian/plugins/koji/__init__.py:109]: Good idea, done

**Review by pholica — APPROVED** [review]

**Comment by pholica** [comment]
Feel free to merge this PR when appropriate/ready. Unfortunately I don't have capacity now for this decision.

**Pull request merged**